aioredlock/algorithm.py

```diff
@@ -119,7 +120,11 @@ def log(self):
 
             raise
 
-        return Lock(self, resource, lock_identifier, valid=True)
+        lock = Lock(self, resource, lock_identifier, lock_timeout, valid=True)
+        if lock_timeout == -1:
+            self.watchdogs[lock.resource] = asyncio.ensure_future(self.__auto_extend(lock))
+
+        return lock
 
     async def extend(self, lock):
```

Comment:
May be add optional timeout argument also here? If not passed use lock's own `lock_timeout`.

Reply:
I think i's not so necessary,but, yeah, add optional arg has no side effects